### PR TITLE
`SideNav` - Deprecate component (HDS-3848)

### PR DESCRIPTION
**Pull request by KristinLBradley**
### :pushpin: Summary

If merged, this PR will deprecate the HDS `SideNav` component.

* **Web Docs Preview:** https://hds-website-git-hds-3848-sidenav-deprecation-hashicorp.vercel.app/components/side-nav
  - I added deprecations warnings to the top of both the "Guidelines" and "Code" tabs.
 
**Related branch:**

* `AddSide` Component: https://github.com/hashicorp/design-system/pull/2384

### :link: External links

* Jira ticket: [HDS-3848](https://hashicorp.atlassian.net/browse/HDS-3848)

***

### 👀 Component checklist

- [x] Percy was checked for any visual regression
- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://hashicorp.atlassian.net/wiki/spaces/HDS/pages/3243114706/Changelog+authoring+best+practices#Templates))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-3848]: https://hashicorp.atlassian.net/browse/HDS-3848?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/src/components/hds/side-nav/index.ts [hunk 1]
```diff
@@ -67,11 +68,26 @@ export default class HdsSideNav extends Component<HdsSideNavSignature> {
...
+    deprecate(
+      'The `Hds::SideNav` component is now deprecated and will be removed in the next major version of `@hashicorp/design-system-components`. Use `Hds::AppSideNav` instead.',
+      false,
+      {
+        id: 'hds.components.sidenav',
+        until: '5.0.0', // TODO: update with actual version number
+        url: 'https://helios.hashicorp.design/components/side-nav?tab=version%20history#4110', // TODO: make sure URL matches correct version number
+        for: '@hashicorp/design-system-components',
+        since: {
+          enabled: '4.11.0', // TODO: check if this is the correct version
```
Comment (KristinLBradley) [inline packages/components/src/components/hds/side-nav/index.ts:87]: Need to make sure I have the right version numbers for all these...

**Review by jorytindall — APPROVED** [review]
To me this looks good, but I don't have broader knowledge/context for how this is generally done in Ember.

**Comment by MelSumner** [comment]
It looks good but seems like it needs a rebase and for a few questions answered about version numbers. 👍

**Review by MelSumner — APPROVED** [review]
Conditionally approve since it looks correct to me. If we decide to deprecate this with the new component release, this PR seems gtg. 👍

File: packages/components/src/components/hds/side-nav/index.ts (hunk 1 above)
Comment (alex-ju) [inline packages/components/src/components/hds/side-nav/index.ts:87]:
```suggestion
          enabled: '4.14.0',
```

File: website/docs/components/side-nav/index.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ navigation:
   hidden: false
   keywords: ['navigation', 'side navigation', 'sidenav', 'sidebar']
 status:
-  updated: 4.10.0
+  deprecated: 4.11.0
```
Comment (alex-ju) [inline website/docs/components/side-nav/index.md:14]:
```suggestion
  deprecated: 4.14.0
```

**Review by alex-ju — APPROVED** [review]
Gave this a fresh look. We need to do some updates on the version number, otherwise looks good

**Pull request merged**